cmd/oras/pull.go

```diff
@@ -148,9 +148,28 @@ func runPull(opts pullOptions) error {
 		return ret, nil
 	}
 
+	ctx, _ := opts.SetLoggerLevel()
+	var dst = file.New(opts.Output)
+	dst.AllowPathTraversalOnWrite = opts.PathTraversal
+	dst.DisableOverwrite = opts.KeepOldFiles
+
 	pulledEmpty := true
 	copyOptions.PreCopy = display.StatusPrinter("Downloading", opts.Verbose)
 	copyOptions.PostCopy = func(ctx context.Context, desc ocispec.Descriptor) error {
+		// restore named but deduplicated successor nodes
+		successors, err := content.Successors(ctx, dst, desc)
+		if err != nil {
+			return err
+		}
+		for _, s := range successors {
+			if _, ok := s.Annotations[ocispec.AnnotationTitle]; ok {
+				if _, ok := printed.Load(generateKey(s)); !ok {
```

Comment:
Curious on the reason why we use `Load` over `LoadOrStore`.

@wangxiaoxuan273 @qweeah Could you explain a bit?

Reply:
Printed is for deduplication. File target has already done the deduplication since it won't allow duplidated filename, not mentioning duplicated digest+filename.

Reply:
Changed back to `LoadOrStore`.